README.md

```diff
@@ -378,7 +378,15 @@ const useStoreWithUndo = create<StoreState>()(
 
 ### Cool-off period
 
-`handleSet?: (handleSet: StoreApi<TState>['setState']) => StoreApi<TState>['setState']`
+```typescript
+  handleSet?: (handleSet: StoreApi<TState>['setState']) => (
+    pastState: Parameters<StoreApi<TState>['setState']>[0],
+    // `replace` will likely be deprecated and removed in the future
+    replace: Parameters<StoreApi<TState>['setState']>[1],
+    currentState: PartialTState,
+    deltaState?: Partial<PartialTState> | null,
+) => void
+```
```

Comment:
I have some concerns that this type definition may spook new users reading the README, but it also seems like a good idea to be accurate here?

Reply:
I'm not sure. The use of the `Parameters` utility type is actually easier on the eyes to me than directly using/copying/extending the types from `SetStateInternal` from zustand:

```typescript
type SetStateInternal<T> = {
    _(partial: T | Partial<T> | {
        _(state: T): T | Partial<T>;
    }['_'], replace?: boolean | undefined): void;
}['_'];
```
I think the addition in this PR just looks ugly (in part?) because, whereas previously `replace` was hidden within `StoreApi<TState>['setState']`, we make it visible here in order to add the two additional parameters `currentState` and `deltaState`

Reply:
I agree. Let's merge this in as is, but I'll make a follow-up PR with some options.